src/handleTestResults.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { writeFile, mkdir } from 'node:fs/promises';
+import { AgentTestDetailsResponse, jsonFormat, humanFormat, junitFormat } from '@salesforce/agents';
+import { Ux } from '@salesforce/sf-plugins-core/Ux';
+
+async function writeFileToDir(outputDir: string, fileName: string, content: string): Promise<void> {
+  // if directory doesn't exist, create it
+  await mkdir(outputDir, { recursive: true });
+
+  await writeFile(`${outputDir}/${fileName}`, content);
+}
+
+export async function handleTestResults({
+  id,
+  format,
+  results,
+  jsonEnabled,
+  outputDir,
+}: {
+  id: string;
+  format: 'human' | 'json' | 'junit';
+  results: AgentTestDetailsResponse | undefined;
+  jsonEnabled: boolean;
+  outputDir?: string;
+}): Promise<void> {
+  if (!results) {
+    // do nothing since there are no results to handle
+    return;
```

Comment:
Will an error be thrown in other code?

Reply:
No need to throw an error here. The idea is that the command could complete without having results to show (e.g. async test run, timed out resume, etc...)